### PR TITLE
[staging-next] python3Packages.deepdiff: fix build

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

https://hydra.nixos.org/build/141451362
https://github.com/NixOS/nixpkgs/pull/119398

Adds new requirements, bumps the version, exposes as application.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/clevercsv/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    tabview
+  ];
+
+  pythonImportsCheck = [
+    "clevercsv"
+    "clevercsv.cparser"
+  ];
+
+  # tests cannot import clevercsv package
+  doCheck = false;
```
Comment (mweinelt) [inline pkgs/development/python-modules/clevercsv/default.nix:40]:
```
running install tests
test_consistency (unittest.loader._FailedTest) ... ERROR

======================================================================
ERROR: test_consistency (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_consistency
Traceback (most recent call last):
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/build/source/tests/test_unit/test_consistency.py", line 12, in <module>
    from clevercsv.consistency import get_best_set
  File "/build/source/clevercsv/__init__.py", line 7, in <module>
    from .cparser_util import field_size_limit
  File "/build/source/clevercsv/cparser_util.py", line 10, in <module>
    from .cparser import Parser, Error as ParserError
ModuleNotFoundError: No module named 'clevercsv.cparser'


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
```

Looking for ideas here.

Comment (Mic92) [reply pkgs/development/python-modules/clevercsv/default.nix]: Fixed. see comment.

File: pkgs/development/python-modules/clevercsv/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  ];
+
+  pythonImportsCheck = [
+    "clevercsv"
+    "clevercsv.cparser"
+  ];
+
+  checkPhase = ''
+    # by linking the installed version the tests also have access to compiled native libraries
+    rm -rf clevercsv
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/clevercsv/default.nix:41]:
```suggestion
    rm -r clevercsv
```

Normally -f is not required and hides if the directory changes.

**Comment by mweinelt** [comment]
Thanks, squashed the commits and added you as co-author.

**Pull request merged**